source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -0,0 +1,372 @@
+                                                                      const double temperature,
+                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
+                                                                      const DislocationCreepParameters dislocation_creep_parameters,
+                                                                      const PeierlsCreepParameters peierls_creep_parameters,
+                                                                      const double min_viscosity,
+                                                                      const double max_viscosity) const
+      {
+        std::pair<double, double> creep_edot_and_deriv = std::make_pair(0., 0.);
+
+        if (use_diffusion_creep)
+          creep_edot_and_deriv = creep_edot_and_deriv + diffusion_creep->compute_strain_rate_and_derivative(creep_stress, pressure, temperature, diffusion_creep_parameters);
+
+        if (use_dislocation_creep)
+          creep_edot_and_deriv = creep_edot_and_deriv + dislocation_creep->compute_strain_rate_and_derivative(creep_stress, pressure, temperature, dislocation_creep_parameters);
+
+        if (use_peierls_creep)
+          creep_edot_and_deriv = creep_edot_and_deriv + peierls_creep->compute_strain_rate_and_derivative(creep_stress, pressure, temperature, peierls_creep_parameters);
+
+        const double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*creep_edot_and_deriv.first;
+        const double strain_rate_deriv = 1./(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*creep_edot_and_deriv.second;
+
+        return std::make_pair(strain_rate, strain_rate_deriv);
+      }
+
+
+
+      template <int dim>
+      void
+      CompositeViscoPlastic<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.declare_entry ("Include diffusion creep", "true",
+                           Patterns::Bool (),
+                           "Whether to include diffusion creep in the rheological formulation.");
+
+        prm.declare_entry ("Include dislocation creep", "true",
+                           Patterns::Bool (),
+                           "Whether to include dislocation creep in the rheological formulation.");
+
+        prm.declare_entry ("Include Peierls creep", "true",
+                           Patterns::Bool (),
+                           "Whether to include Peierls creep in the rheological formulation.");
+
+        // Diffusion creep parameters
+        Rheology::DiffusionCreep<dim>::declare_parameters(prm);
+
+        // Dislocation creep parameters
+        Rheology::DislocationCreep<dim>::declare_parameters(prm);
+
+        // Dislocation creep parameters
+        Rheology::PeierlsCreep<dim>::declare_parameters(prm);
+
+        // Some of the parameters below are shared with the rheology models,
```

Comment:
```suggestion
        // Some of the parameters below are shared with the material models,
```
or perhaps I'm misreading? Will the values declared here override the values in the material models?

Reply:
No, the values declared here override only the values in the subordinate rheology models (diffusion, dislocation, ...). 
Most of the time, this will not matter - if the user declares a value for a parameter in the prm file, then that value will be used. These comments are only for people reading the code - the user shouldn't notice any odd behaviour.

I've extended the comment to be more explicit.

Reply:
👍